src/spclient.rs

```diff
@@ -323,15 +346,23 @@ impl SpClient {
                 subtype: PSBT_SP_SUBTYPE,
                 key: PSBT_SP_ADDRESS_KEY.as_bytes().to_vec(),
             }) {
-                // Create the right output key
                 let sp_address = SilentPaymentAddress::try_from(deserialize::<String>(value)?)?;
-                let output_key = silentpayments::sending::generate_recipient_pubkey(
-                    sp_address.into(),
-                    partial_secret,
-                )?;
-                // update the script pubkey
-                output.script_pubkey =
-                    ScriptBuf::new_p2tr_tweaked(output_key.dangerous_assume_tweaked());
+                if let Some(xonlypubkeys) = sp_address2xonlypubkeys.get_mut(&sp_address.to_string())
+                {
+                    if !xonlypubkeys.is_empty() {
+                        let output_key = xonlypubkeys.remove(0); // actually we could randomize it
```

Comment:
What are the advantages of randomizing it? I haven't checked but i assume right now the addresses are given in the order of `n` ascending, so it's not lexicographic or anything

Reply:
Hhhmm I was probably thinking about randomizing the order of the outputs in the transaction, so it has nothing to do with it and won't happen here anyway. I'll remove the comment